src/CBLURLEndpointListener_Internal.hh

```diff
@@ -0,0 +1,150 @@
+
+    uint16_t port() const {
+        if (_port == 0 && _c4listener) {
+            _port = _c4listener->port();
+        }
+        return _port;
+    }
+
+    FLMutableArray _cbl_nullable getUrls() const {
+        if (!_c4listener) return (FLMutableArray) nullptr;
+
+        CBLDatabase* cblDb = CBLCollection_Database(_conf.collections[0]);
+        auto urls = fleece::MutableArray::newArray();
+        cblDb->c4db()->useLocked([&](C4Database* db) {
+            for ( const std::string& url : _c4listener->URLs(db, kC4SyncAPI) )
+                urls.append(url);
+        });
+        return (FLMutableArray)FLValue_Retain(urls);
+    }
+
+    CBLConnectionStatus getConnectionStatus() const {
+        if (_c4listener) {
+            auto [total, active] = _c4listener->connectionStatus();
+            return {total, active};
+        } else {
+            return {0, 0};
+        }
+    }
+
+    bool start() {
+        if (_c4listener) return true;
+
+        bool succ = true;
+        slice dbname;
+        DEFER {
+            if (!succ) {
+                for (C4Collection* coll: _collections) {
+                    // These collections have been successfully shared.
+                    (void)_c4listener->unshareCollection(dbname, coll);
+                }
+                (void)_c4listener->unshareDB(_db);
+                delete _c4listener;
+                _c4listener = nullptr;
+                _collections.clear();
+                _db = nullptr;
+            }
+        };
+
+        Assert(_conf.collectionCount > 0);
+
+        C4ListenerConfig c4config {
+            _conf.port,
+            _conf.networkInterface,
+            kC4SyncAPI
+        };
+        c4config.allowPush = true;
+        c4config.allowPull = !_conf.readOnly;
+        c4config.enableDeltaSync = _conf.enableDeltaSync;
+
+        _c4listener = new C4Listener(c4config);
+
+        CBLDatabase* cblDb = CBLCollection_Database(_conf.collections[0]);
+        cblDb->c4db()->useLocked([&](C4Database* db) {
+            dbname = db->getName();
+            if (_c4listener->shareDB(dbname, db)) {
+                _db = db;
+            }
+            if (_db) {
+                for (unsigned i = 0; i < _conf.collectionCount; ++i) {
+                    _conf.collections[i]->useLocked([&](C4Collection* coll) {
+                        if (_c4listener->shareCollection(dbname, coll)) {
+                            _collections.push_back(coll);
+                        }
+                    });
+                    if (_collections.size() < i + 1) break;
+                }
+            }
+        });
+        return (succ = _collections.size() == _conf.collectionCount);
+    }
+
+    void stop() {
+        if (!_c4listener) return;
+
+        auto dbname = _db->getName();
+        for (C4Collection* coll: _collections) {
+            (void)_c4listener->unshareCollection(dbname, coll);
+        }
+        (void)_c4listener->unshareDB(_db);
```

Comment:
Are these needed? I think iOS doesn't do this.

Reply:
Oh, no, they are already deleted in the latest commit. I think it's outdated.